codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/protocols/parse/EventStreamUnmarshallerGeneratorTest.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.client.smithy.protocols.parse
+
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.ArgumentsProvider
+import org.junit.jupiter.params.provider.ArgumentsSource
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.rust.codegen.client.testutil.testSymbolProvider
+import software.amazon.smithy.rust.codegen.core.smithy.CodegenTarget
+import software.amazon.smithy.rust.codegen.core.smithy.generators.builderSymbol
+import software.amazon.smithy.rust.codegen.core.smithy.protocols.parse.EventStreamUnmarshallerGenerator
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestModels
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestTools
+import java.util.stream.Stream
+
+class UnmarshallTestCasesProvider : ArgumentsProvider {
+    override fun provideArguments(context: ExtensionContext?): Stream<out Arguments> =
+        EventStreamTestModels.TEST_CASES.map { Arguments.of(it) }.stream()
+}
```

Comment:
Perhaps in `EventStreamTestTools`?

Reply:
Same goes for `software.amazon.smithy.rust.codegen.server.smithy.protocols.serialize.MarshallTestCasesProvider` and `software.amazon.smithy.rust.codegen.client.smithy.protocols.serialize.MarshallTestCasesProvider`.

Reply:
Although in the server we don't support `ec2Query` or `awsQuery`, and we very badly support `restXml`, so maybe we want to filter those out in the server too.

Although then again, if these specific event stream tests pass in the server I guess there's no harm in leaving them in so as to not regress in case we want to support those protocols in the future, so I'd just make the test case providers identical for both subprojects.

Reply:
I kept these argument providers in `codegen-client` and `codegen-server` to avoid needing to depend on junit in `codegen-core`. However, if we set up a `codegen-core-test` module, then that won't be a problem anymore.

Reply:
I played around with creating `codegen-core-test`, and it seems like it _might be_ the right path forward long-term. However, it's a large scale refactor that has rippling effects that will require making `codegen-client-test` a Kotlin module so that `sdk-codegen` can depend on test-only code from `codegen-client`.

To do it nicely, we're going to have to move all of `codegen-core`'s unit tests into `codegen-core-test/src/test`, while test code that is intended to be shared will need to reside in `codegen-core-test/src/main`. I was able to create and populate `codegen-core-test` easily enough, but then code in `codegen-client/src/main` no longer compiles due to requiring test-only code in `src/main` so that `sdk-codegen` can reuse that client-specific test-only code.

One consequence of going through with this is that there will be no way to unit test code that is `internal`, and if we decide to reuse the existing `codegen-client-test` and `codegen-server-test` modules, they will now run both unit tests and codegen tests (but maybe that's desirable?).

I think I'm going to punt on this for this PR.